drivers/clk/clk-divider.c

```diff
@@ -0,0 +1,383 @@
+/****************************************************************************
+ * drivers/clk/clk-divider.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/clk/clk.h>
+#include <nuttx/clk/clk-provider.h>
+
+#include <debug.h>
+#include <stdlib.h>
+
+#include "clk.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define to_clk_divider(_clk) (struct clk_divider *)(_clk->private_data)
```

Comment:
```suggestion
#define to_clk_divider(_clk) (FAR struct clk_divider *)(_clk->private_data)
```
Missing `FAR` qualifier, required to be used in generic code.